dask/array/core.py

```diff
@@ -1659,11 +1667,22 @@ def __setitem__(self, key, value):
             self.dask = y.dask
             self._name = y.name
             self._chunks = y.chunks
-            return self
-        else:
-            raise NotImplementedError(
-                "Item assignment with %s not supported" % type(key)
-            )
+            return
```

Comment:
Why is the `self` removed?

Reply:
The return value from `__setitem__` is arbitrary, and so should therefore be `None`. I have checked that this is how other libraries use it (`numpy`, `xarray`, etc). 

This change should have no consequence, as the return value is never used nor accessible, but makes things more consistent.

Reply:
Is this true even in the case of

```
A[i] = B[i] = C[i]
```

Reply:
... I think so ... at least insofar as the current code appears to behave like numpy:

```python
In [28]: import numpy as np

In [29]: import dask.array as da

In [30]: a = np.array([1, 2, 3])

In [31]: b = np.array([4, 5, 6])

In [32]: c = np.array([7, 8, 9])

In [33]: a[0] = b[0] = c[0]

In [34]: a, b, c
Out[34]: (array([7, 2, 3]), array([7, 5, 6]), array([7, 8, 9]))

In [35]: d = da.from_array(np.array([1, 2, 3]))

In [36]: e = da.from_array(np.array([4, 5, 6]))

In [37]: f = da.from_array(np.array([7, 8, 9]))

In [38]: d[0] = e[0] = f[0]

In [39]: da.compute(d, e, f)
Out[39]: (array([7, 2, 3]), array([7, 5, 6]), array([7, 8, 9]))

```